kinopoisk/movie/sources.py

```diff
@@ -279,11 +279,29 @@ class MovieTrailersPage(KinopoiskPage):
     Parser of kinopoisk trailers page
     """
     url = '/film/{id}/video/'
+    base_trailer_url = 'gettrailer.php?quality=hd&trailer_id={}'
 
     def parse(self):
-        trailers = re.findall(r'GetTrailerPreview\(([^\)]+)\)', self.content)
-        for trailer in trailers:
-            self.instance.add_trailer(json.loads(trailer.replace("'", '"')))
-
-        self.instance.youtube_ids = list(set(re.findall(r'//www.youtube.com/v/(.+)\?', self.content)))
-        self.instance.set_source(self.content_name)
+        # Because some films have different URL and ID
+        current_film_url = re.findall(
+            r'href="https://www.kinopoisk.ru/film/.+/video',
+            self.content
+        )
+
+        current_film_id = current_film_url[0].split('/')[-2:-1][0]
+        trailers_kinopoisk_urls = list(set(
+            re.findall(r'/film/{}/video/\d+'.format(current_film_id),
+                       self.content)
+        ))
+
+        for trailer_id in trailers_kinopoisk_urls:
+            trailer_id = trailer_id.split('/')[-1:][0]
+            pure_trailer_url = self.base_trailer_url.format(trailer_id)
+            self.instance.add_trailer({
```

Comment:
Are you sure there are only 2 fields available now? Also, it seems `file` always could be generated from `id`, am I right? If yes is an answer to both questions, I don't see any reason to have other fields in Trailer class.

Reply:
... and `file` could be specified as 
```
class Trailer
  @property
  def file(self):
    return ...
```

Reply:
Yes, we can parse only URL for tralier.
Yes, always generated from `id`.